guides/source/association_basics.md

```diff
@@ -140,210 +236,198 @@ create_table :books do |t|
 end
 ```
 
-### The `has_one` Association
+This ensures that even if `optional: true` allows `author_id` to be NULL, when
+it is not NULL, it must point to a valid record in the authors table.
```

Comment:
I found this sentence a bit hard to parse at first. Here's an attempt at a rewrite:

```suggestion
This ensures that even if `optional: true` is set, allowing `author_id` to be NULL, when
it the value is not NULL, it must point to a valid record in the authors table.
```

Reply:
Thanks @MatheusRich, there was a typo in the suggested text, but I used a variation of it to make it easier to read ✨